### ISSUE TITLE
Creating and calling 2 models one after another

### ISSUE THREAD

**Issue by aidonchuk** [NVIDIA/retinanet-examples#121]
Hi, have a problem crashing inference when 2 models created and executed one after another
If i comment model1 inference every think goes ok. Models exported by cppapi
`import random
import time

import torch
import torch.nn.functional as F
from PIL import Image

from retinanet.model import get_model_trt

def infer_trt(im_path, resize, max_size):
    model = get_model_trt(path='../models/retinanet_rn18fpn_1_stage.engine')
    model1 = get_model_trt(path='../models/retinanet_rn34fpn_2_stage.engine')

    im = Image.open(im_path).convert("RGB")

    if isinstance(resize, list):
        resize = random.randint(resize[0], resize[-1])

    ratio = resize / min(im.size)
    if ratio * max(im.size) > max_size:
        ratio = max_size / max(im.size)
    im = im.resize((int(ratio * d) for d in im.size), Image.BILINEAR)

    data = get_data(im, model)
    data1 = get_data(im, model1)

    with torch.no_grad():
        print(data.shape)
        # for i in range(0, 50):
        start_time = time.time()
        scores, boxes, classes = model(data)
        print("--- %s seconds ---" % (time.time() - start_time))
        scores, boxes, classes = model1(data1)
        print("--- %s seconds ---" % (time.time() - start_time))
        print(scores, boxes, classes)

def get_data(im, model):
    mean = [0.485, 0.456, 0.406]
    std = [0.229, 0.224, 0.225]

    data = torch.ByteTensor(torch.ByteStorage.from_buffer(im.tobytes()))
    data = data.float().div(255).view(*im.size[::-1], len(im.mode))
    data = data.permute(2, 0, 1)

    for t, mean, std in zip(data, mean, std):
        t.sub_(mean).div_(std)

    # Apply padding
    pw, ph = ((model.stride - d % model.stride) % model.stride for d in im.size)
    data = F.pad(data, (0, pw, 0, ph))
    data = data.unsqueeze(0).cuda()
    return data

infer_trt('asd.png', 256, 512)
`

**Comment by aidonchuk** [comment]
If two models use same engine everything ok.

**Comment by aidonchuk** [comment]
Something like this - don't work too.
'import random
import time

import torch
import torch.nn.functional as F
from PIL import Image

from retinanet.trt_model import get_model_trt


def infer_trt(im_path, resize, max_size):
    model = get_model_trt(path='../models/retinanet_rn18fpn_1_stage.engine')
    #model1 = get_model_trt(path='../models/retinanet_rn34fpn_2_stage.engine')

    im = Image.open(im_path).convert("RGB")

    if isinstance(resize, list):
        resize = random.randint(resize[0], resize[-1])

    ratio = resize / min(im.size)
    if ratio * max(im.size) > max_size:
        ratio = max_size / max(im.size)
    im = im.resize((int(ratio * d) for d in im.size), Image.BILINEAR)

    data = get_data(im.copy(), model)
    #data1 = get_data(im.copy(), model1)

    with torch.no_grad():
        print(data.shape)
        # for i in range(0, 50):
        start_time = time.time()
        scores, boxes, classes = model(data)
        print("--- %s seconds ---" % (time.time() - start_time))
        del model
        model = get_model_trt(path='../models/retinanet_rn34fpn_2_stage.engine')
        data = get_data(im.copy(), model)
        scores, boxes, classes = model(data)
        print("--- %s seconds ---" % (time.time() - start_time))
        print(scores, boxes, classes)


def get_data(im, model):
    mean = [0.485, 0.456, 0.406]
    std = [0.229, 0.224, 0.225]

    data = torch.ByteTensor(torch.ByteStorage.from_buffer(im.tobytes()))
    data = data.float().div(255).view(*im.size[::-1], len(im.mode))
    data = data.permute(2, 0, 1)

    for t, mean, std in zip(data, mean, std):
        t.sub_(mean).div_(std)

    # Apply padding
    pw, ph = ((model.stride - d % model.stride) % model.stride for d in im.size)
    data = F.pad(data, (0, pw, 0, ph))
    data = data.unsqueeze(0).cuda()
    return data

infer_trt('asd.png', 256, 512)
'

**Comment by yashnv** [comment]
What is `get_model_trt`? If you have added code, that is not supported, then open a pull request. This is not an issue, closing this.

**Comment by aidonchuk** [comment]
@yashnv Please don't harry so much
`def get_model_trt(path):
    return Engine.load(path)`
